### DISCUSSION TITLE
Navigation Framework based on Cicerone

### DISCUSSION THREAD

**Discussion by terrakok** [terrakok/Cicerone#139] [Ideas]
I've noticed that lots of people want to extend Cicerone for product cases e.g.:
 - add multi backstack fragment with tabs to library
 - add deeplink handler to library
 - add some base fragments for fast development Single Activity Application to library

But I want to leave Cicerone lightweight, easy and extendable library but not framework.

If someone wants to create powerful navigation framework based on Cicerone I will support it with development, ideas and promotion :)

**Comment by terrakok** [comment]
My experiments with navigation framework based on Cicerone you can find in [Fondazion](https://github.com/terrakok/Cicerone/tree/fondazione/fondazione/src/main/java/com/github/terrakok/fondazione).  

It contains:
 - AppActivity - base application class for Single Activity App with app router
 - AppFragment - base fragment with additional useful lifecycle callbacks
 - NavigationFragment - fragment with navigation container and inner router

I think it may be good baseline for building navigation framework.

**Reply by ultraon** [reply]
Still didn't find an example with deeplinks + Cicerone
